pkg/authenticator/ecrsecret.go

```diff
@@ -0,0 +1,134 @@
+package authenticator
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	varPackagesNamespace = "eksa-packages"
+	varConfigMapName     = "ns-secret-map"
+	varECRTokenName      = "ecr-token"
+)
+
+type ecrSecret struct {
+	clientset    kubernetes.Interface
+	nsReleaseMap map[string]string
+}
+
+var _ Authenticator = (*ecrSecret)(nil)
+
+func NewECRSecret(config *rest.Config) (*ecrSecret, error) {
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	cm, err := clientset.CoreV1().ConfigMaps(varPackagesNamespace).
+		Get(context.TODO(), varConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return &ecrSecret{
+		clientset:    clientset,
+		nsReleaseMap: cm.Data,
+	}, nil
+}
+
+func (s *ecrSecret) AuthFilename() string {
+	// Check if Helm registry config is set
+	helmconfig := os.Getenv("HELM_REGISTRY_CONFIG")
+	if helmconfig != "" {
+		// Use HELM_REGISTRY_CONFIG
+		return helmconfig
+	}
+
+	return ""
+}
+
+func (s *ecrSecret) UpdateConfigMap(ctx context.Context, name string, namespace string, add bool) error {
+	cm, err := s.clientset.CoreV1().ConfigMaps(varPackagesNamespace).
+		Get(ctx, varConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	addToConfigMap(cm, name, namespace, add)
+	_, err = s.clientset.CoreV1().ConfigMaps(varPackagesNamespace).
+		Update(ctx, cm, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	// Store data
+	s.nsReleaseMap = cm.Data
+
+	return nil
+}
+
+func addToConfigMap(cm *v1.ConfigMap, name string, namespace string, add bool) {
```

Comment:
The `add` parameter here is misleading. If it's set to false, then what happens is the value is removed. That conflicts with the name of the function.

Break this into two functions, named for what they do. 

If you really have to have one function, call it `modifyConfigMap` and pass an "operation" enum value of either ADD or REMOVE or something.